Top up canister (#1384)
dfinity/portal: docs/developer-docs/production/topping-up-canister.md

```bash
dfx canister --network ic status jqylk-byaaa-aaaal-qbymq-cai
```

Output will look like this:

```bash
Canister status call result for jqylk-byaaa-aaaal-qbymq-cai.
Status: Running
Controllers: mto6d-zfnut-rlsxr-ogdeg-apo53-evpob-ljgnp-ma2x3-6yf3b-t4rd5-qqe t5j57-vyaaa-aaaal-qatsq-cai
Memory allocation: 0
Compute allocation: 0
Freezing threshold: 2_592_000
Memory Size: Nat(2471918)
Balance: 9_811_813_913_485 Cycles
Module hash: 0xe7866e1949e3688a78d8d29bd63e1c13cd6bfb8fbe29444fa606a20e0b1e33f0
```

### Option 1: If you have ICP on your account

If you have ICP on the account associated with a `dfx` identity, you can tell the Ledger Canister to take some of that ICP, convert it to cycles, and give it to a canister of your choice: `dfx ledger [OPTIONS] top-up --amount <AMOUNT> <DESTINATION>`

`dfx ledger --network ic top-up --amount 0.1 jqylk-byaaa-aaaal-qbymq-cai`

```bash
dfx ledger account-id
dfx ledger --network ic balance
dfx ledger --network ic top-up --amount 0.1 jqylk-byaaa-aaaal-qbymq-cai
```

> -   The `dfx ledger account-id` returns the ledger `account` id of the current `dfx` identity used

Reviewer: ```suggestion
-   The `dfx ledger account-id` returns the ledger account id of the current `dfx` identity used
```

If you have a cycles wallet you control via dfx, you can send cycles from your cycles wallet to a canister of your choice: `dfx canister deposit-cycles <AMOUNT> <DESTINATION>`.

```bash
dfx wallet --network ic balance
dfx canister deposit-cycles 1000000 jqylk-byaaa-aaaal-qbymq-cai 
```

-   The `wallet --network ic balance` checks the cycles balance of your cycles wallet on mainnet
-   The `canister deposit-cycles` takes cycles from your cycles wallet and gives them to the canister of your choice

### Special Case: topping up another cycles wallet

Cycles wallets are canisters like any other, so you can top them up as well. If you have a cycles wallet you control via dfx, there is another option as well for sending cycles from your cycles wallet to a canister of your choice: `dfx wallet send [OPTIONS] <DESTINATION> <AMOUNT>`.

In this example, we assume we are sending cycles to a cycles wallet with principal `dfds-sddds-aaaal-qbsms-cai`.

```bash
dfx wallet --network ic balance
dfx wallet --network ic send 1000000 dfds-sddds-aaaal-qbsms-cai 
```

-   The `wallet --network ic balance` checks the cycles balance of your cycles wallet on mainnet

> -   The `wallet --network ic send 1000000 dfds-sddds-aaaal-qbsms-cai` takes 1000000 cycles from your cycles wallet and send sthem to cycles wallet `dfds-sddds-aaaal-qbsms-cai`.

Reviewer: ```suggestion
-   The `wallet --network ic send 1000000 dfds-sddds-aaaal-qbsms-cai` takes 1000000 cycles from your cycles wallet and sends them to cycles wallet `dfds-sddds-aaaal-qbsms-cai`.
```

# Topping up & Refilling a Canister with Cycles

A common pattern for developer workflow is to refill canisters with cycles so they can continue to operate as intended.

## Basic Rules 

1. Anyone can top up any canister deployed to the Internet Computer (does not need to be the author or controller)
2. One can top up a canister via `dfx`, [NNS Frontend dapp](https://nns.ic0.app), or third-party service (e.g. https://cycleops.dev). All one needs is the canister's principal.
3. There are a few ways to top up canisters so this article goes through each one.

## Topping up a canister with `dfx`

In the following examples, we will try to top up canister `jqylk-byaaa-aaaal-qbymq-cai` from the [first tutorial](../../tutorials/deploy_sample_app.md) with a million cycles (1000000). These instructions can work for any canister principal or any cycles amount.

### Number of cycles

> For easy of use, you can copy/paste the cycles amounts below:

Reviewer: ```suggestion
For ease of use, you can copy/paste the cycles amounts below:
```

```bash
dfx canister --network ic status jqylk-byaaa-aaaal-qbymq-cai
```

Output will look like this:

```bash
Canister status call result for jqylk-byaaa-aaaal-qbymq-cai.
Status: Running
Controllers: mto6d-zfnut-rlsxr-ogdeg-apo53-evpob-ljgnp-ma2x3-6yf3b-t4rd5-qqe t5j57-vyaaa-aaaal-qatsq-cai
Memory allocation: 0
Compute allocation: 0
Freezing threshold: 2_592_000
Memory Size: Nat(2471918)
Balance: 9_811_813_913_485 Cycles
Module hash: 0xe7866e1949e3688a78d8d29bd63e1c13cd6bfb8fbe29444fa606a20e0b1e33f0
```

### Option 1: If you have ICP on your account

If you have ICP on the `account` associated with a `dfx` identity, you can tell the Ledger Canister to take some of that ICP, convert it to cycles, and give it to a canister of your choice: `dfx ledger [OPTIONS top-up --amount <AMOUNT> <DESTINATION>`

`dfx ledger --network ic top-up --amount 0.1 jqylk-byaaa-aaaal-qbymq-cai`

```bash
dfx ledger account-id
dfx ledger --network ic balance
dfx ledger --network ic top-up --amount 0.1 jqylk-byaaa-aaaal-qbymq-cai
```

> -   The `dfx dfx ledger account-id` returns the `account` id of the current `dfx` identity used

Reviewer: ```suggestion
-   The `dfx ledger account-id` command returns the `account` id of the current `dfx` identity used
```

-   The `dfx ledger --network ic balance` command checks how much balance is on the `account` associated with the current `dfx` identity used
-   `--network ic` tells `dfx` to use the mainnet IC as the network (not anything local for example)
-   `top-up --amount 0.1 jqylk-byaaa-aaaal-qbymq-cai` command converts 0.1 ICP into cycles and uses them to refill canister `jqylk-byaaa-aaaal-qbymq-cai`.

### Option 2: If you have cycles on your cycles wallet

If you have a cycles wallet you control via dfx, you can send cycles from your cycles wallet to a canister of your choice: `dfx canister deposit-cycles <AMOUNT> <DESTINATION>`.

```bash
dfx wallet --network ic balance
dfx canister deposit-cycles 1000000 jqylk-byaaa-aaaal-qbymq-cai 
```

-   The `wallet --network ic balance` checks the cycles balance of your cycles wallet on mainnet
-   The `canister deposit-cycles` takes cycles from your cycles wallet and gives them to the canister of your choice

I want top up canister `jqylk-byaaa-aaaal-qbymq-cai` with a million cycles (1000000).

### Special Case: topping up another cycles wallet

> Cycles wallets are canisters like any other, so you can top them up as well. If you have a cycles wallet you control via dfx, there is another option as well for sending cycles from your cycles wallet to a canister of your choice: `dfx wallet send [OPTIONS] <DESTINATION> <AMOUNT>`.

Reviewer: The header mentions that it is only used for topping up another cycles wallet, but the body disagrees  ('canister of your choice' would be misleading here)

Author: you are right. will fix
